src/components/search/index.tsx

```diff
@@ -0,0 +1,29 @@
+import { useState } from 'react';
+import styled from 'styled-components';
+import { getSearchResults } from '../../services/product.service';
+import { Button } from '../general/button';
+import { Input } from '../general/input';
+
+const StyledButton = styled.div`
+  width: 20%;
+`;
+
+export const SearchInput: React.FC = () => {
+  const [searchTerm, setSearchTerm] = useState('');
+  const handleClick = () => {
+    getSearchResults(searchTerm).then((data) => console.log(data));
```

Comment:
Have a look at how to use async/await syntax. It's a bit cleaner. Don't forget to handle potential errors :)